### ISSUE TITLE
crack hash new create database

### ISSUE THREAD

**Issue by Slait** [defuse/crackstation-hashdb#10]
I was able to create a database
php createidx.php md5 words.txt words-md5.idx

Then I sorted it out
./sortidx -r 4048 words-md5.idx

Checked via test.php

But how do I specify the hash to check, for example: apple (md5: 1F3870BE274F6C49B3E31A0C6728957F)

php test.php md5
Successfully cracked [apple].
Successfully cracked [apple] (as partial match).

How to specify 1F3870BE274F6C49B3E31A0C6728957F to find an apple?

**Comment by defuse** [comment]
Good point, the only way to do that currently is to write a script that uses LookupTable.php like test.php does. I should add a script that takes in hashes on the command line. It might be a while before I have time to do that so if you want to do it I'll gladly accept accept a pull request.

**Comment by markgc61** [comment]
How do I get cookies

**Comment by defuse** [comment]
here's an okay cookie recipe: https://www.wholesomeyum.com/recipes/low-carb-keto-cream-cheese-cookies/

**Comment by jamesyoungdigital** [comment]
Hi Slait, wrote a little program to do just that.

Please check my repo, it's identical but contains test/lookup_hash.php.  It needs some improvements but it works.  I'll update with more as I go along here.

**Comment by jamesyoungdigital** [comment]
> Good point, the only way to do that currently is to write a script that uses LookupTable.php like test.php does. I should add a script that takes in hashes on the command line. It might be a while before I have time to do that so if you want to do it I'll gladly accept accept a pull request.

When I clean the program up, could you accept a pull request for this issue?

**Comment by markgc61** [comment]
I have a friend in England who writes his own accounts. No hacking.. just
wrote a code that he won't tell me.. if maybe we gang up on him hahaha, no
but he's a step ahead.
That's where I get my TV viewing
$70 a yr per.
Get three and prices drop

**Comment by jamesyoungdigital** [comment]
So you're saying a decent re-write in C++ (why an entire rewrite?) instead of incremental improvements for test cases, and replacements of just the PHP scripts?

I suppose I could re-write everything in C++ and try a few other methods instead of using crackstation's hashdb, but if it works very well and it does, I wonder why I would reinvent the wheel so to speak.

Perhaps... unless I use several commodity machines and e.g. Apache's Hadoop File System and other crazy experiments, just for the heck of it!

**Comment by jamesyoungdigital** [comment]
> Good point, the only way to do that currently is to write a script that uses LookupTable.php like test.php does. I should add a script that takes in hashes on the command line. It might be a while before I have time to do that so if you want to do it I'll gladly accept accept a pull request.

I've written one.  It's hack-ish but it does just that.

**Comment by markgc61** [comment]
If it all new accounts no hacking maybe he is using a vcc

**Comment by skyzedout** [comment]
> 
> 
> Hi Slait, wrote a little program to do just that.
> 
> Please check my repo, it's identical but contains test/lookup_hash.php. It needs some improvements but it works. I'll update with more as I go along here.

Thanks for the release man,
Perhaps do you know why when using "sortidx" with a large file (+2GB) ,i instantly get  "Index sort complete" (when in reality it doesn't sorted anything), or "Not a valid index file"? ,
Currently i splitted the original "words.txt" files in smaller ones,but can't understand why i get this issue,and after all i'm not a c++ veteran :)
